src/main/rp/RP.ts

```diff
@@ -98,9 +123,22 @@ export class RP {
     const verifyAuthenticationResponseOpts = this.newVerifyAuthorizationResponseOpts({
       ...opts,
     });
-    const authorizationResponse = await AuthorizationResponse.fromPayload(authorizationResponsePayload);
-
-    return await authorizationResponse.verify(verifyAuthenticationResponseOpts);
+    const authorizationResponse = await AuthorizationResponse.fromPayload(authorizationResponsePayload).catch((error: Error) => {
+      rpEventEmitter.emit(AuthorizationEvents.ON_AUTH_RESPONSE_RECEIVED_FAILED, authorizationResponsePayload, error);
```

Comment:
I would expect event objects, which then already create the timestamp of the event.

So create a generic event object with a subject that is a generic. Then in this example set the responsePayload as the subject. The event uses the current date when it is created. Then for handlers it means they do not have to generate timestamps. Also the event would have the error placeholder. So you can always have a method on the event hasError for instance